src/tr2/game/ui/dialogs/graphic_settings.c

```diff
@@ -27,7 +27,71 @@ typedef struct {
     int32_t misc;
 } M_OPTION;
 
-static M_OPTION m_Options[] = {
+typedef struct {
+    int32_t value;
+    GAME_STRING_ID name;
+} M_ENUM_ENTRY;
+
+typedef struct {
+    const M_ENUM_ENTRY *entry;
+    int32_t position;
+    int32_t count;
+} M_ENUM_LOOKUP;
+
+static const M_ENUM_ENTRY m_TextureFilterOptions[] = {
+    // clang-format off
+    { GFX_TF_NN, GS_ID(MISC_OFF), },
+    { GFX_TF_BILINEAR, GS_ID(DETAIL_BILINEAR), },
+    { -1, nullptr, },
+    // clang-format on
+};
+
+static const M_ENUM_ENTRY m_LightingContrastOptions[] = {
+    // clang-format off
+    { LIGHTING_CONTRAST_LOW, GS_ID(DETAIL_LIGHTING_CONTRAST_LOW), },
+    { LIGHTING_CONTRAST_MEDIUM, GS_ID(DETAIL_LIGHTING_CONTRAST_MEDIUM), },
+    { LIGHTING_CONTRAST_HIGH, GS_ID(DETAIL_LIGHTING_CONTRAST_HIGH), },
+    { -1, nullptr, },
+    // clang-format on
+};
+
+static const M_ENUM_ENTRY m_RenderModeOptions[] = {
+    // clang-format off
+    { RM_SOFTWARE, GS_ID(DETAIL_RENDER_MODE_SOFTWARE), },
+    { RM_HARDWARE, GS_ID(DETAIL_RENDER_MODE_HARDWARE), },
+    { -1, nullptr, },
+    // clang-format on
+};
+
+static const M_ENUM_ENTRY m_AspectModeOptions[] = {
+    // clang-format off
+    { AM_4_3, GS_ID(DETAIL_ASPECT_MODE_4_3), },
+    { AM_16_9, GS_ID(DETAIL_ASPECT_MODE_16_9), },
+    { AM_ANY, GS_ID(DETAIL_ASPECT_MODE_ANY), },
+    { -1, nullptr, },
+    // clang-format on
+};
+
+// Indexed by M_OPTION.misc
+static const M_ENUM_ENTRY *m_EnumOptions[] = {
+    m_TextureFilterOptions,
+    m_LightingContrastOptions,
+    m_RenderModeOptions,
+    m_AspectModeOptions,
+    nullptr,
+};
```

Comment:
I believe we can get by without this enum if we choose to make `misc` a `void*` general pointer, that for the current entities will be cast to `(int32_t)(intptr_t)`, and for the newly introduced enums, it'll let them reference `M_ENUM_ENTRY[]` directly.

Reply:
Thanks, done 😌